groovy/gradle/src/org/netbeans/modules/gradle/GradleDataObject.java

```diff
@@ -57,16 +59,24 @@
         extension = {"gradle"},
         position = 290
 )
-@DataObject.Registration(
+@DataObject.Registrations({
+        @DataObject.Registration(
         mimeType = GradleDataObject.MIME_TYPE,
         iconBase = "org/netbeans/modules/gradle/resources/gradle.png",
         displayName = "#LBL_GradleFile_LOADER",
         position = 300
-)
+    ),
+    @DataObject.Registration(
+        mimeType = GradleDataObject.KOTLIN_MIME_TYPE,
```

Comment:
Possibly you could handle this without special "gradle+kotlin" MIME type. Just register your own `DataObject.Factory` into "kotlin-only" MIME type and position it before the regular `KotlinDataObject`.

Reply:
Thank you for the tip, I'm going to try that, so I won't have to modify the FS code.

Reply:
Well it seems the default declarative MIME resolver is more powerful than documented at first sight. So it is possible to do the recognition without this change.
Moreover it is also possible to convert the Dockerfile MIME resolver to the declarative one. I'm going to do that in another PR along with some improvement on the documentation.

Reply:
Finally could get rid of this extra declaration matching on kts extension and .gradle.kts in the filename. Not perfect, but good enough.
The custom mime-type remained as it we registering a custom "Run Gradle" action on that.